File: src/test/java/no/fintlabs/repository/utils/performance/DurationFormatter.java
package no.fintlabs.repository.utils.performance;

import java.time.Duration;
import java.util.StringJoiner;
import java.util.function.Consumer;

public class DurationFormatter {
    public static String formatDuration(Duration duration) {
        StringJoiner stringJoiner = new StringJoiner(" ");
        performIfPositive(duration.toMinutes(), v -> stringJoiner.add(String.format("%2dm", v)));
        performIfPositive(duration.toSecondsPart(), v -> stringJoiner.add(String.format("%2ds", v)));
        performIfPositive(duration.toMillisPart(), v -> stringJoiner.add(String.format("%3dms", v)));
        return stringJoiner.toString();
    }

    private static void performIfPositive(long value, Consumer<Long> valueConsumer) {
        if (value > 0) {
            valueConsumer.accept(value);
        }
    }
}

File: src/test/kotlin/no/novari/flyt/history/repository/utils/performance/DurationFormatter.kt
package no.novari.flyt.history.repository.utils.performance

import java.time.Duration
import java.util.StringJoiner

object DurationFormatter {
    @JvmStatic
    fun formatDuration(duration: Duration): String {
        val stringJoiner = StringJoiner(" ")
        performIfPositive(duration.toMinutes()) { value -> stringJoiner.add(String.format("%2dm", value)) }
        performIfPositive(duration.toSecondsPart().toLong()) { value -> stringJoiner.add(String.format("%2ds", value)) }
        performIfPositive(duration.toMillisPart().toLong()) { value -> stringJoiner.add(String.format("%3dms", value)) }
        return stringJoiner.toString()
    }

    private fun performIfPositive(
        value: Long,
        valueConsumer: (Long) -> Unit,
    ) {
        if (value > 0) {
            valueConsumer(value)
        }
    }
}
